### PR TITLE
Switch to trunk based development & explicit semantic versioning

### PR DESCRIPTION
**Pull request by rtertiaer**
This PR updates our release documentation. This came up because of some oddities during the stamping of the 0.3.0 release candidate 0. 

I'm trying to more closely follow a [more conventional trunk-based development](https://trunkbaseddevelopment.com/), where there exists only one trunk and version branches never get merged back into `main`. As the doc was before this PR, we'd at least be merging compiled javascript back into `main` regularly, which is no bueno. I think it's also a bit simpler and cleaner.

### CODE REVIEW

File: NEW_RELEASE.md [hunk 1]
```diff
@@ -4,10 +4,10 @@ So you think you are ready to make a release, eh? Follow the steps below :)
 
 ## Making a prelease
 - [ ] Merge main into develop
-- [ ] Make a branch off develop called VERSION-prelease
+- [ ] Make a trunk branch off develop using the version number, ie `0.3.0`
```
Comment (Lohrer) [inline NEW_RELEASE.md:7]: Shouldn't this be a version branch? The _trunk_ branch for us is `main` unless I'm misunderstanding your proposed new workflow.

Comment (rtertiaer) [reply NEW_RELEASE.md]: I've gone slash & burn with this doc; have a gander at the latest version :)

**Comment by linknum23** [comment]
Given our constraint of needing to check in "compiled" javascript for the
next couple of releases I think this makes sense from a high level.

**Comment by rtertiaer** [comment]
@linknum23 @Lohrer is there further discussion to be had here, or is this approved/denied?

I ask cuz I just merged https://github.com/micro-nova/AmpliPi/pull/575 into the wrong trunk :upside_down_face:

File: NEW_RELEASE.md [hunk 2]
```diff
@@ -34,11 +21,19 @@ So you think you are ready to make a release, eh? Follow the steps below :)
...
-- [ ] merge the prerelease branch into develop and merge that into the main branch
-- [ ] use poetry to bump the version `poetry version ${VERSION} && git add pyproject.toml && git commit -m "Bump release"`
-- [ ] use git to delete any prelease tags `git push origin -d $(git tag -l "${VERSION}-alpha*") && git tag -d $(git tag -l "${VERSION}-alpha*")`
-- [ ] Tag the changes so we can make a release on GitHub `git tag -as ${VERSION} -m '' && git push --follow-tags`
-- [ ] Make a release using the GitHub interface, add notes from the CHANGELOG.md
-- [ ] Make a new Raspberry Pi image to program onto units before shipping.
+## Versioning
+This project follows [Semantic Versioning](https://semver.org/). Here are some examples of versions we've used before:
+* `0.2.1`: this is a public general-availability bugfix release of the `0.2` feature release.
+* `0.3.0-alpha.0` is the first alpha release of the `0.3` feature release.
```
Comment (linknum23) [inline NEW_RELEASE.md:27]: probably add beta and rc examples as well

**Comment by linknum23** [comment]
The only issue here is that some documentation that links to github may be ahead of the current release. We may be able to handle those things on a case by case basis though.

**Review by linknum23 — APPROVED** [review]

**Comment by rtertiaer** [comment]
> The only issue here is that some documentation that links to github may be ahead of the current release. We may be able to handle those things on a case by case basis though.

there are two ways of handling this I think. one, we could generate docs on a version-by-version basis, like a drop down menu up top of a generated site. additionally, putting a note besides each new feature or edgecase that we program for stating which version it was introduced or changed is very helpful. you can see both examples of this setup in [the Python docs](https://docs.python.org/3/library/os.html#os.environb) (the bits that say "_New in 3.9_", and the dropdown up top.) All that's required of us is to regularly maintain our manual with changes (hopefully the PR checklist helps) and one day get good at generating documentation sites, instead of just leaning on github (but that's definitely a future TODO.)

File: NEW_RELEASE.md [hunk 3]
```diff
@@ -34,11 +21,19 @@ So you think you are ready to make a release, eh? Follow the steps below :)
...
+
+## Making a release
+- [ ] Ensure the PR(s) with your features & fixes are merged into `main`.
+- [ ] Check out and pull `main`
+- [ ] Update the changelog with notes and the latest version (and commit it!)
```
Comment (Lohrer) [inline NEW_RELEASE.md:32]: This reads as if the changelog/notes should be committed directly onto main, but that's not possible right?

Comment (rtertiaer) [reply NEW_RELEASE.md]: ope, good catch. I think the latest rendition of this oughta be "more correct".

**Review by Lohrer — APPROVED** [review]
Seems reasonable to me, but I've never created a release so I'd suggest waiting on the wisdom of Lincoln.

**Pull request merged**